Sources/StreamChatUI/ChatMessageList/MessageComposer/ChatMessageComposerCommandCollectionViewCell_Tests.swift

```diff
@@ -2,25 +2,94 @@
+    private let defaultCellSize: CGSize = CGSize(width: 250, height: 50)
+    
+    /// Default content of the view
+    private let defaultCommand = Command(name: "Default", description: "", set: "", args: "[text]")
+
     func test_defaultAppearance() {
-        let view = ChatMessageComposerCommandCellView()
-        view.command = .init(name: "Hello", description: "what", set: "is this", args: "WHAT")
-        AssertSnapshot(view, size: CGSize(width: 300, height: 40), record: true)
+        let cell = ChatMessageComposerCommandCollectionViewCell()
+        cell.commandView.content = defaultCommand
+        AssertSnapshot(cell, size: defaultCellSize)
     }
 
-    func test_defaultAppearance_withDirectMessageChannel() {}
+    func test_defaultAppearance_allAvailableCommands() {
+        let cell = ChatMessageComposerCommandCollectionViewCell()
+        cell.uiConfig.images.commandIcons.forEach { command in
+            cell.commandView.content = .init(name: command.key, description: "", set: "", args: "[text]")
+            AssertSnapshot(cell, variants: [.defaultLight], size: defaultCellSize, suffix: command.key)
+        }
+    }
 
-    func test_defaultAppearance_withNonDMChannel() {
-        // TODO: https://stream-io.atlassian.net/browse/CIS-652
+    func test_appearanceCustomization_usingUIConfig() {
+        class CustomView: ChatMessageComposerCommandCellView {
+            override func setUpAppearance() {
+                super.setUpAppearance()
+                
+                backgroundColor = .systemGray
+                commandNameSubtitleLabel.textColor = UIColor.white
+                commandNameLabel.textColor = UIColor.darkGray
+            }
+            
+            override func setUpLayout() {
+                super.setUpLayout()
+                
+                commandNameSubtitleLabel
+                    .trailingAnchor
+                    .constraint(equalTo: layoutMarginsGuide.trailingAnchor)
+                    .isActive = true
+            }
+        }
+        
+        let cell = ChatMessageComposerCommandCollectionViewCell()
+        cell.uiConfig.messageComposer.suggestionsCommandCellView = CustomView.self
+        cell.commandView.content = defaultCommand
+        
+        AssertSnapshot(cell, variants: [.defaultLight], size: defaultCellSize)
     }
 
-    func test_appearanceCustomization_usingUIConfig() {}
+    func test_appearanceCustomization_usingAppearanceHook() {
+        class CustomView: ChatMessageComposerCommandCollectionViewCell {}
+        CustomView.defaultAppearance {
+            $0.commandView.backgroundColor = .systemGray
+            $0.commandView.commandNameSubtitleLabel.textColor = UIColor.white
+            $0.commandView.commandNameLabel.textColor = UIColor.darkGray
+        }
+        
+        let view = CustomView()
+        view.commandView.content = defaultCommand
+        AssertSnapshot(view, variants: [.defaultLight], size: defaultCellSize)
```

Comment:
Why only `defaultLight`? What if subclassing breaks dynamic fonts? 😱

Reply:
I was trying to reduce the test duplication since all variants are already tested in previous tests. How would subclassing break the dynamic fonts? 🤔 Unless we overrode the UILabel and removed the adjust property, but that would be a client's issue. The way I see it is that subclassing and appearance hook tests should be focusing on testing if the customization works. If dynamic fonts or light/dark modes are compromised is probably something that the client is doing wrong? Or maybe I'm missing something? 🤔

Reply:
All right, I'm probably too paranoid 👍 Leave it as is